gridmet_aggregation_PRMS.py

```diff
@@ -0,0 +1,105 @@
+    gridmet_drb_gdf = gpd.GeoDataFrame(gridmet_drb_df)
+
+    return gridmet_drb_gdf
+
+# gridmet_prms_area_avg_agg() function aggregates to groupby_cols scale and outputs the aggregated dataset
+def gridmet_prms_area_avg_agg(df, groupby_cols, val_colnames, wgt_col, output_path = None):
+    """
+    :param pd.DataFrame or gpd.GeoDataFrame df:
+    :param  list groupby_cols: group by cols of df
+    :param list val_colnames: list of columns to perform area-weighted average calculation on
+    :param str wgt_col: weight column
+    :param str output_path: if not None, path to save output df dataset in local dir.
+    :return: df with gridmet metrics per PRMS_segid
+    """
+
+    ## Weighted average calc = sum((val[1] * weight[1]), ... , (val[n] * weight[n]) ) / sum(weight[1],...,weight[n])
+
+    start = time.perf_counter()
+
+    df = df.copy()
+    ## 1. Calc value x weight col
+    for col in val_colnames:
+           df[f'{col}_x_area'] = df[col]*df[wgt_col]
+
+    ## 2. Create group by, by summing new value cols weighted by area
+    df_grouped = df.groupby(groupby_cols).agg(area_m2_sum = (wgt_col, 'sum'),
+                                              tmmx_wgtd_sum = ('tmmx_x_area', 'sum'),
+                                              tmmn_wgtd_sum = ('tmmn_x_area', 'sum'),
+                                              pr_wgtd_sum = ('pr_x_area', 'sum'),
+                                              srad_wgtd_sum = ('srad_x_area', 'sum'),
+                                              vs_wgtd_sum = ('vs_x_area', 'sum'),
+                                              rmax_wgtd_sum = ('rmax_x_area', 'sum'),
+                                              rmin_wgtd_sum = ('rmin_x_area', 'sum'),
+                                              sph_wgtd_sum = ('sph_x_area', 'sum')
+                                              )
+    ## 3. Get weighted average
+    for col in val_colnames:
+        df_grouped[f'{col}_wgtd_avg'] = df_grouped[f'{col}_wgtd_sum'] / df_grouped['area_m2_sum']
+
+    # remove interim col
+    df_final = df_grouped.loc[:, ~(df_grouped.columns.str.contains('wgtd_sum'))]
+
+    if output_path is not None:
+        df_final.to_csv(output_path, sep = ',')
+        print('Output saved in: ' + output_path)
+
+    end = time.perf_counter()
+
+    print('Time (sec) ellapsed:', end - start)
+
+    return df_final
+
+# Define variables and run
+if __name__ =='__main__':
+
+    ## Variable definitions
+    gdf_prms_path_edited = './data/GFv1_catchments_edited.gpkg'
+    gdf = gpd.read_file(gdf_prms_path_edited, layer = 'GFv1_catchments_edited')
+    gridmet_ncdf = './data/t_climate_2022_03_09.nc'
+    data_vars_shrt_all = ['tmmx', 'tmmn', 'pr', 'srad', 'vs', 'rmax', 'rmin', 'sph']
+
+    ## Create dataframe and merge with shapefile information
+    gridmet_drb_gdf = ncdf_to_gdf(ncdf_path=gridmet_ncdf,
+                                 shp = gdf,
+                                 left_on = 'geomid',
```

Comment:
Could `geomid` be changed to something that is a more universal identifier, like the `hru_id_nat` from the National Geospatial Fabric? As far as I understand `geomid` is a zero-based index using the rows of the `.gpkg`, and this is used as the identifier in the `.nc` file that contains the HRU-based gridMET data (e.g. the `.nc` files [here](https://doimspp.sharepoint.com/sites/usgs-GS-WMA-IWP-PUMP/Shared%20Documents/Forms/AllItems.aspx?id=%2Fsites%2Fusgs%2DGS%2DWMA%2DIWP%2DPUMP%2FShared%20Documents%2F40%2E2%20Data%2Ddriven%20salinity%2FData%2FDRB%2DInland%2FgridMET&CT=1648579049600&OR=OWA%2DNT&CID=7ca64cda%2Ddf57%2Dbb13%2D069f%2D7c82c2562f3f)). But it's not really a stand-alone ID so that `.nc` file with HRU-based data isn't very useful when trying to extract data based on HRUs. 

I am trying to extract gridMET drivers based on HRUs to update a PRMS-SNTemp run (which requires HRU-based drivers), and I'm not sure how to cross-walk between the `geomid` and the National Geospatial Fabric `hru_id_nat`

Reply:
I see an `hru_id` in the `GFv1_catchments_edited.gpkg` file that we are using (ranges from 2846 to 4790). Would that work, @jzwart?

Reply:
Gotcha. That's helpful. Can you confirm that the `hru_id_nat` is the identifier that you want, @jzwart?

If that's the case, we should do that cross-walk from the `nhruNationalIdentifier` layer in the national GF file and I think that step should happen upstream in the `drb-network-prep` repo.

Reply:
> But it's not really a stand-alone ID so that .nc file with HRU-based data isn't very useful when trying to extract data based on HRUs.

@jzwart, in the regridding process, the geometry attribute is removed from the final `.nc` file. I have go back to the regridding process to figure out why this happened, but that is why I merge it back to geodataframe (using geomid) for plotting or for the aggregation to csv. 

> RE: Changing geomid to a more universal identifier: 

Alternatively, to avoid uncovering an available cross-walk, or building one, are you suggesting we simply modify the colname `geomid` to a more readable name @jzwart ? This way it is more clear why that col is used to merge the geodataframe attributes.

Reply:
@msleckman  yes, merging on a `hru_id_nat` would be great, if that's easy enough. Having the `hru_id_nat` in that `.nc` file would make it very easy to use. 

@jsadler2 agreed that the ID's should be updated in the `drb-network-prep` repo rather than here. And yes the `hru_id_nat` is what I'd need to run PRMS easily.

Reply:
Made an issue for this here: https://github.com/USGS-R/drb-network-prep/issues/6